plugins/repoclosure.py

```diff
@@ -92,6 +92,10 @@ def _get_unresolved(self, arch=None):
                 for pkgs_filtered in available.filter(reponame=repo):
                     checkpkgs.add(pkgs_filtered)
             pkgs.intersection_update(checkpkgs)
+            # --best not applied earlier due to --check, so do it now
+            if self.base.conf.best:
+                available = available.latest()
+                available.run()
```

Comment:
the line here ``available.run()`` is completely  not necessary. it returns list of packages from query. Why did you used it here?

Reply:
I wanted to enumerate the package list, which doesn't happen until either you iterate over the query or use run(). The package list needs to be enumerated by the time unresolved_deps is calculated, and it wouldn't happen without the run().

Reply:
I guess that ``available = available.latest().apply()`` would be better.

Reply:
I see here several problem from my PR:
line 75:
``available.latest()`` has to be replaced with ``available = available.latest()``
line 77:
``available.filter(latest=True)`` has to be replaced with ``available = available.filter(latest=True)``

Reply:
Where is `apply()` documented?

Reply:
Regarding the problems with your PR, this is what I was getting at in the last part of rpm-software-management/dnf-plugins-extras/pull/50#issuecomment-264415495.

Reply:
Unfortunately query apply() is not documented anywhere. Query.run() is mentioned in hawkey documentation, but hawkey is not alive anymore. Libdnf that replaces hawkey doesn't have documentation yet.

Reply:
Using `apply()` does seem to work for me, though I don't have a "failing" condition yet so I can't say for certain.